### PR TITLE
Add sample to show disabling listener behaviour selectively.

### PR DESCRIPTION
**Pull request by krmahadevan**
Fix for issue https://github.com/cbeust/testng/issues/972

### CODE REVIEW

**Comment by krmahadevan** [comment]
@cbeust @juherr 
Can you please help get this merged ?

File: doc/documentation-main.html [hunk 1]
```diff
@@ -2033,6 +2033,42 @@ <h3 class="sourcetitle">Shell</h3>
 
 Note that the <tt>@Listeners</tt> annotation will apply to your entire suite file, just as if you had specified it in a <tt>testng.xml</tt> file. If you want to restrict its scope (for example, only running on the current class), the code in your listener could first check the test method that's about to run and decide what to do then.
 
+Here's how it can be done.
+
+<ol>
+  <li>First define a new custom annotation that can be used to specify this restriction.
+
```
Comment (cbeust) [inline doc/documentation-main.html]: Colon at the end of this sentence.

File: doc/documentation-main.html [hunk 2]
```diff
@@ -2033,6 +2033,42 @@ <h3 class="sourcetitle">Shell</h3>
...
+      <pre class="brush: java">
+      @Retention(RetentionPolicy.RUNTIME)
+      @Target ({ElementType.TYPE})
+      public @interface DisableListener {}
+      </pre>
+
+  <li>Add an edit check as below within your regular listeners
```
Comment (cbeust) [inline doc/documentation-main.html]: Colon.

File: doc/documentation-main.html [hunk 3]
```diff
@@ -2033,6 +2033,42 @@ <h3 class="sourcetitle">Shell</h3>
...
+          ConstructorOrMethod consOrMethod =iInvokedMethod.getTestMethod().getConstructorOrMethod();
+          DisableListener disable = consOrMethod.getMethod().getDeclaringClass().getAnnotation(DisableListener.class);
+          if (disable != null) {
+              return;
+          }
+          //else resume your normal operations
+      }
+      </pre>
+
+  <li>Annotate test classes wherein the listener is not to be invoked as shown below
```
Comment (cbeust) [inline doc/documentation-main.html]: Colon and no need for "as shown below".

**Comment by cbeust** [comment]
Sure thing, just a few minor comments.

**Comment by krmahadevan** [comment]
Thanks Cedric. I will come back from work in the evening and address the comments and update this PR. 

-Krishnan Mahadevan

"All the desirable things in life are either illegal, expensive, fattening or in love with someone else"

-----Original Message-----

**Comment by krmahadevan** [comment]
@cbeust 
I have addressed your review comments. Please let me know if its fine.

**Comment by cbeust** [comment]
Looks good, thanks @krmahadevan !

**Pull request merged**